todo-app/src/stores/TodosStore.js

```diff
@@ -0,0 +1,172 @@
+import { observable, action, runInAction } from "mobx"
+
+
+
+
+
+const createTaskFormDefault = {
+    description: "",
+    deadline: ""
+}
+
+const API_ROOT = 'https://hyf-react-api.herokuapp.com'
+
+class TodosStore {
+
+    @observable
+    todosList = []
+
+    @observable
+    createTaskForm = createTaskFormDefault
+
+    @observable
+    editingDescriptionText = ""
+
+    @observable
+    editingDeadlineTime = ""
+
+    @observable
+    taskEditingId = null
+
+    @action
+    getTodos = async () => {
+        const res = await fetch(`${API_ROOT}/todos`)
+        const parsedRes = await res.json()
+        console.log(parsedRes)
+        runInAction(() => {
+            this.todosList = parsedRes
+        })
+    }
+    @action
+    createTodo = async () => {
+        if (this.createTaskForm.description && this.createTaskForm.deadline) {
+            await fetch(`${API_ROOT}/todos/create`, {
+                method: "POST",
+                headers: {
+                    "content-type": "application/json"
+                },
+                body: JSON.stringify(this.createTaskForm)
+            })
+            runInAction(() => {
+                // reset the form
+                this.createTaskForm = createTaskFormDefault
+            })
+        } else {
+            alert("you should fill the values of Description and Deadline")
+        }
+        this.getTodos()
```

Comment:
I think your `this.getTodos()` call should be inside your `if` block, not here at the end. You're making API calls even when the user hasn't input the correct fields yet.